ihe-interop/src/test/java/org/motechproject/ihe/interop/handler/helper/IHEActionHelperTest.java

```diff
@@ -61,4 +68,24 @@ public void shouldThrowExceptionWhenRecipientNotFound() throws ClassNotFoundExce
         when(iheTemplateDataService.findByName("sampleTemplate3")).thenReturn(cdaTemplate);
         iheActionHelper.handleAction(params);
     }
+
+    @Test
+    public void shouldParseLargeTemplate() throws ClassNotFoundException, ParserConfigurationException, SAXException, IOException {
+        Map<String, Object> params = new HashMap<>();
+        params.put(Constants.TEMPLATE_NAME_PARAM, "sampleTemplate4");
+        params.put(Constants.RECIPIENT_NAME_PARAM, "sampleRecipient4");
+
+        byte[] bytes = IOUtils.toByteArray(getClass().getResourceAsStream("/large_template_sample.txt"));
+        Byte[] byteObjects = new Byte[bytes.length];
+        int i = 0;
+        for (byte b : bytes) {
+            byteObjects[i++] = b;
+        }
+
+        when(iheTemplateDataService.findByName("sampleTemplate4")).thenReturn(cdaTemplate);
```

Comment:
use real xml template file, this test fails with previous template because its not xml, we should have a situation when running this test with old template throws java too large method error